workers/data_refinery_workers/downloaders/array_express.py

```diff
@@ -33,20 +30,16 @@ def download_array_express(job_id):
         utils.end_job(job, None, False)
         return
 
-    target_directory = ROOT_URI + batch.internal_location
-    os.makedirs(target_directory, exist_ok=True)
-
-    filename = batch.download_url.split('/')[-1]
-    target_file_name = target_directory + filename
+    target_file_path = utils.prepare_destination(batch)
 
     logger.info("Downloading file from %s to %s. (Batch #%d, Job #%d)",
                 batch.download_url,
-                target_file_name,
+                target_file_path,
                 batch.id,
                 job_id)
 
     try:
-        target_file = open(target_file_name, "wb")
+        target_file = open(target_file_path, "wb")
         request = requests.get(batch.download_url, stream=True)
 
         for chunk in request.iter_content(CHUNK_SIZE):
```

Comment:
Sounds good. Mostly wanted to be sure we were thinking about these things here, but also hadn't seen the `CHUNK_SIZE` code before and wondered why it was there.